src/message/__tests__/fetchActions-test.js

```diff
@@ -342,46 +342,33 @@ describe('fetchActions', () => {
       test("rejects when validation-at-the-edge can't handle data, dispatches MESSAGE_FETCH_ERROR", async () => {
         // This validation is done in transformFetchedMessages in
         // rawModelTypes.
-        //
-        // Simulate #4156, a real-life problem that a user at server commit
-        // 0af2f9d838 ran into [1], by having `user` be missing on reactions
-        // on a message:
-        //   https://github.com/zulip/zulip-mobile/issues/4156#issuecomment-655905093
         const store = mockStore<GlobalState, Action>(baseState);
 
-        // Missing `user` (and `user_id`, for good measure), to
-        // simulate #4156.
-        const faultyReaction = {
-          reaction_type: 'unicode_emoji',
-          emoji_code: '1f44d',
-          emoji_name: 'thumbs_up',
-        };
-
         const response = {
-          // Flow would complain at `faultyReaction` if it
+          // Flow would complain at `avatar_url` if it
           // type-checked `response`, but we should ignore it if that
-          // day comes. It's badly shaped on purpose.
-          messages: [{ ...fetchedMessage1, reactions: [faultyReaction] }],
+          // day comes. It's badly typed on purpose.
+          messages: [{ ...fetchedMessage1, avatar_url: 123 }],
```

Comment:
From reading the avatars code (since that's now the only part of `transformFetchedMessages` that seems like it could throw an error on bad server data), here's a different fun way to cause an exception:
```js
          messages: [{ ...fetchedMessage1, avatar_url: null, sender_email: undefined }],
```

That causes the `email.toLowerCase()` here to throw:
```js
        `${GravatarURL.ORIGIN}/avatar/${md5(email.toLowerCase())}?d=identicon`,
```
and doesn't provoke any `logging` calls.

(The `sender_email: undefined` is a crude way to get the same effect in the consuming code as if `sender_email` were omitted altogether. Could also use a line or two more code here and do that properly.)